packages/adblocker/src/filters/network.ts

```diff
@@ -879,29 +930,34 @@ export default class NetworkFilter implements IFilter {
       mask,
 
       // Optional parts
-      csp: (optionalParts & 1) === 1 ? buffer.getNetworkCSP() : undefined,
       filter:
-        (optionalParts & 2) === 2
+        (optionalParts & 1) === 1
           ? isUnicode
             ? buffer.getUTF8()
             : buffer.getNetworkFilter()
           : undefined,
-      hostname: (optionalParts & 4) === 4 ? buffer.getNetworkHostname() : undefined,
-      domains: (optionalParts & 8) === 8 ? Domains.deserialize(buffer) : undefined,
-      rawLine: (optionalParts & 16) === 16 ? buffer.getRawNetwork() : undefined,
-      redirect: (optionalParts & 32) === 32 ? buffer.getNetworkRedirect() : undefined,
-      denyallow: (optionalParts & 64) === 64 ? Domains.deserialize(buffer) : undefined,
+      hostname: (optionalParts & 2) === 2 ? buffer.getNetworkHostname() : undefined,
+      domains: (optionalParts & 4) === 4 ? Domains.deserialize(buffer) : undefined,
+      rawLine: (optionalParts & 8) === 8 ? buffer.getRawNetwork() : undefined,
+      denyallow: (optionalParts & 16) === 16 ? Domains.deserialize(buffer) : undefined,
+      modifierOptionValue:
+        (optionalParts & 32) === 32
+          ? getBit(mask, NETWORK_FILTER_MASK.isCSP)
+            ? buffer.getNetworkCSP()
+            : getBit(mask, NETWORK_FILTER_MASK.isRedirect)
+              ? buffer.getNetworkRedirect()
+              : buffer.getUTF8()
+          : undefined,
```

Comment:
I think it might make sense to replace both `getNetworkCSP` and `getNetworkRedirect` with a unified `getModifierOptionValue` (including computing the codebooks of all these values instead of having separate one).

Reply:
I was thinking of that, but in a view of backward compatibility, I think we can leave the method as-is.

Reply:
However, I think integrating the codebook won't be a problem here.

Reply:
@seia-soto if that helps, we could split this PR into two parts:
1. introduce `modifierOptionValue`
2. implement `$replace` on top of it